modules/objdetect/src/qrcode.cpp

```diff
@@ -1245,5 +1228,1208 @@ cv::String QRCodeDetector::detectAndDecode(InputArray in,
+          int pixel = bin_barcode.at<uchar>(indices[r].y , indices[r].x);
+          if(pixel == 255)
+          {
+              count_w++;
+          }
+          if(pixel == 0)
+          {
+              count_b++;
+          }
+      }
+     double frac = (double)count_b / (double)count_w;
+     if ((frac <= 0.835) || (frac >= 1.24)) {return false;}
+     return true;
+}
+
+bool QRDetectMulti::checkPointsInsideQuadrangle(vector<Point2f> triangle_points, vector<Point2f> all_localization_points)
+{
+      Point2f p0(triangle_points[0].x, triangle_points[0].y);
+      Point2f p1(triangle_points[1].x, triangle_points[1].y);
+      Point2f p2(triangle_points[2].x, triangle_points[2].y);
+      Point2f p3(triangle_points[3].x, triangle_points[3].y);
+      Mat lineMask = Mat::zeros(bin_barcode.size(), bin_barcode.type());
+      line( lineMask, p0, p1, 255, 1, 8, 0);
+      line( lineMask, p1, p2, 255, 1, 8, 0);
+      line( lineMask, p2, p3, 255, 1, 8, 0);
+      line( lineMask, p0, p3, 255, 1, 8, 0);
+      vector< vector< Point > > contours;
+      vector<Vec4i> hierarchy;
+      findContours(lineMask, contours, hierarchy, RETR_TREE, CHAIN_APPROX_SIMPLE);
+      int count = 0;
+      if(contours.size() != 2) return false;
+      for(size_t i = 0; i < all_localization_points.size(); i++)
+      {
+            double dist;
+            if((dist = pointPolygonTest(contours[0], all_localization_points[i], false)) > 0)
+            {
+               count++;
+            }
+       }
+       if (count == 3) return true;
+       else return false;
+}
+
+bool QRDetectMulti::checkPointsInsideTriangle(vector<Point2f> triangle_points, vector<Point2f> all_localization_points)
+{
+      Point2f p0(triangle_points[0].x, triangle_points[0].y);
+      Point2f p1(triangle_points[1].x, triangle_points[1].y);
+      Point2f p2(triangle_points[2].x, triangle_points[2].y);
+      Mat lineMask = Mat::zeros(bin_barcode.size(), bin_barcode.type());
+      line( lineMask, p0, p1, 255, 1, 8, 0);
+      line( lineMask, p1, p2, 255, 1, 8, 0);
+      line( lineMask, p2, p0, 255, 1, 8, 0);
+      vector< vector< Point > > contours;
+      vector<Vec4i> hierarchy;
+      findContours(lineMask, contours, hierarchy, RETR_TREE, CHAIN_APPROX_SIMPLE);
+      if(contours.size() < 1) return false;
+      int count = 0;
+      double eps = 3;
+      for(size_t i = 0; i < all_localization_points.size(); i++)
+      {
+            double dist;
+            if((dist = pointPolygonTest(contours[0], all_localization_points[i], false)) > 0)
+            {
+
+               if((abs(all_localization_points[i].x - p0.x) > eps) && (abs(all_localization_points[i].x - p1.x) > eps)
+               && (abs(all_localization_points[i].x - p2.x) > eps))
+               {
+                  return false;
+               }
+            }
+       }
+       if (count == 0) return true;
+       else return false;
+}
+
+bool QRDetectMulti::compareSquare::operator()(vector<int> a, vector<int> b) const
```

Comment:
> `vector<int> a`

Do we really need vector here? I see 3 values only.
vectors and other object must be passed by "const reference" by default.